hydrography-approach/processing_scripts/bridge_statistics/create_bridge_stats.py

```diff
@@ -0,0 +1,212 @@
+        raise FileNotFoundError(f"File {gpkg_file} does not exist.")
+    except ValueError as e:
+        logger.error(f"{gpkg_file} is not a valid GeoPackage file: {str(e)}", exc_info=True)
+        raise ValueError(f"{gpkg_file} is not a valid GeoPackage file.")
+    except Exception as e:
+        logger.error(f"Unexpected error while reading GeoPackage file {gpkg_file}: {str(e)}", exc_info=True)
+        raise
+    
+
+def get_csv_length(csv_file: str) -> int:
+    """
+    Reads a GeoPackage file and returns its length.
+
+    Args:
+        csv_file (str): Path to CSV file to read.
+
+    Returns:
+        int: Length of CSV file.
+
+    Raises:
+        FileNotFoundError: If the specified file does not exist.
+        ValueError: If the file is not a valid CSV file.
+    """
+    try:
+        df_csv = pd.read_csv(csv_file)
+        return len(df_csv)
+    except FileNotFoundError as e:
+        logger.error(f"File {csv_file} does not exist: {str(e)}", exc_info=True)
+        raise FileNotFoundError(f"File {csv_file} does not exist.")
+    except ValueError as e:
+        logger.error(f"{csv_file} is not a valid CSV file: {str(e)}", exc_info=True)
+        raise ValueError(f"{csv_file} is not a valid CSV file.")
+    except Exception as e:
+        logger.error(f"Unexpected error while reading CSV file {csv_file}: {str(e)}", exc_info=True)
+        raise
+    
+
+def update_stats(stats: pd.DataFrame, description: str, count: int, stats_list: List[int]) -> Tuple[pd.DataFrame, List[int]]:
+    """
+    Updates the statistics based on the given description and count.
+
+    Args:
+        stats (pd.DataFrame): The DataFrame containing the statistics.
+        description (str): The description of the statistic to be updated.
+        count (int): The count to be appended to the stats list and assigned to the specified description.
+        stats_list (List[int]): The list containing the statistics.
+
+    Returns:
+        Tuple[pd.DataFrame, List[int]]: The updated statistics DataFrame and the updated statistics list.
+
+    Raises:
+        KeyError: If the specified description does not exist in the DataFrame.
+    """
+    try:
+        if not (stats["Description"] == description).any():
+            logger.error(f"Description '{description}' does not exist in the DataFrame.", exc_info=True)
+            raise KeyError(f"Description '{description}' does not exist in the DataFrame.")
+        
+        stats_list.append(count)
+        stats.loc[stats["Description"] == description, "bridges"] = count
+        return stats, stats_list
+    except KeyError as e:
+        logger.error(f"Description '{description}' does not exist in the DataFrame. {str(e)}", exc_info=True)
+        raise KeyError(f"Description '{description}' does not exist in the DataFrame.") from e
+    except Exception as e:
+        logger.error(f"Unexpected error in update_stats: {str(e)}", exc_info=True)
+        raise
+
+
+def create_bridge_statistics(
```

Comment:
The pattern for calculating bridge statistics is repetitive. Consider refactoring it to reduce redundancy. You could encapsulate the logic for calculating and updating statistics in a helper function.

Reply:
Done